fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -141,17 +143,38 @@ def global_extent(self, rank_metadata: QuantityMetadata) -> Tuple[int, ...]:
             rank_metadata.dims, rank_metadata.extent, self.layout
         )
 
-    def subtile_extent(self, global_metadata: QuantityMetadata) -> Tuple[int, ...]:
+    def subtile_extent(self, global_metadata: QuantityMetadata, rank: int, overlap: bool, cube_metadata_offset=0) -> Tuple[int, ...]:
         """Return the shape of a single rank representation for the given dimensions."""
-        return rank_extent_from_tile_metadata(
-            global_metadata.dims, global_metadata.extent, self.layout
+        tile_decomposition = subtile_extents_from_tile_metadata(
+            global_metadata.dims[cube_metadata_offset:], global_metadata.extent[cube_metadata_offset:], self.layout
         )
+        num_decomposed_dims = int(len(tile_decomposition) / 2)
+        horizontal_dim_index = 0
+        return_extent = []
+        for num_dim, dim in enumerate(global_metadata.dims[cube_metadata_offset:]):
+            edge_index_offset = 0
+            is_end_index = False
+            if dim in constants.HORIZONTAL_DIMS:
+                if dim in constants.Y_DIMS:
+                    is_end_index = rank >= (self.layout[horizontal_dim_index] - 1) * self.layout[1 - horizontal_dim_index]
+                    rank_is_on_axis_edge = rank < self.layout[horizontal_dim_index] or is_end_index
+                elif dim in constants.X_DIMS:
+                    position_on_axis = rank % self.layout[horizontal_dim_index]
+                    is_end_index = position_on_axis == self.layout[horizontal_dim_index] - 1
+                    rank_is_on_axis_edge = position_on_axis == 0 or is_end_index
+                if rank_is_on_axis_edge:
+                    edge_index_offset = num_decomposed_dims
+                horizontal_dim_index = horizontal_dim_index + 1
+
+            return_extent.append(_interface_overlap_extent(dim, is_end_index, tile_decomposition[num_dim + edge_index_offset], overlap))
+
+        return tuple(return_extent)
 
     def subtile_slice(
         self,
-        rank: int,
         global_dims: Sequence[str],
         global_extent: Sequence[int],
+        rank: int,
```

Comment:
Please keep call signatures the same where possible.

Reply:
Reverted the argument order change.